From b72103bc43f5076ad0f597d0891e69377ba40ac8 Mon Sep 17 00:00:00 2001
From: Yuxi Sun <b36102@freescale.com>
Date: Tue, 31 Jul 2012 13:44:49 +0800
Subject: [PATCH 338/505] ENGR00218452 v4l2 overlay: add overlay support on
 IPU2

When selected frame buffer is for IPU2, the display frame buffer
name is DISP4 XX, so add judgement when using IPU2

Signed-off-by: Yuxi Sun <b36102@freescale.com>
---
 test/mxc_v4l2_test/mxc_v4l2_overlay.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/test/mxc_v4l2_test/mxc_v4l2_overlay.c b/test/mxc_v4l2_test/mxc_v4l2_overlay.c
index 85fd546..4950e77 100644
--- a/test/mxc_v4l2_test/mxc_v4l2_overlay.c
+++ b/test/mxc_v4l2_test/mxc_v4l2_overlay.c
@@ -630,6 +630,10 @@ main(int argc, char **argv)
 		g_display_lcd = 1;
 	else if (strcmp(fb0_fix.id, "DISP3 BG") == 0)
 		g_display_lcd = 0;
+	else if (strcmp(fb0_fix.id, "DISP4 BG") == 0)
+		g_display_lcd = 3;
+	else if (strcmp(fb0_fix.id, "DISP4 BG - DI1") == 0)
+		g_display_lcd = 4;
 
         if (ioctl(fd_v4l, VIDIOC_S_OUTPUT, &g_display_lcd) < 0)
         {
@@ -665,7 +669,8 @@ main(int argc, char **argv)
 		return TFAIL;
 	}
 
-	if (strcmp(fb_fg_fix.id, "DISP3 FG") != 0) {
+	if (((strcmp(fb_fg_fix.id, "DISP3 FG") != 0) && g_display_lcd < 3)
+		|| ((strcmp(fb_fg_fix.id, "DISP4 FG") != 0) && g_display_lcd > 2)) {
 		close(g_fd_fb_fg);
 		fb_device_fg = "/dev/fb2";
 		if ((g_fd_fb_fg = open(fb_device_fg, O_RDWR)) < 0) {
@@ -680,7 +685,8 @@ main(int argc, char **argv)
 			close(fd_fb_0);
 			return TFAIL;
 		}
-		if (strcmp(fb_fg_fix.id, "DISP3 FG") != 0) {
+		if (((strcmp(fb_fg_fix.id, "DISP3 FG") != 0) && g_display_lcd < 3)
+			|| ((strcmp(fb_fg_fix.id, "DISP4 FG") != 0) && g_display_lcd > 2)) {
 			printf("Cannot find overlay frame buffer\n");
 			close(g_fd_fb_fg);
 			close(fd_fb_0);
-- 
1.8.0

